corehq/motech/auth.py

```diff
@@ -0,0 +1,79 @@
+from datetime import datetime
+
+import jsonobject
+from requests.auth import HTTPBasicAuth, AuthBase, HTTPDigestAuth
+
+from corehq.motech.const import BASIC_AUTH, DIGEST_AUTH, BEARER_AUTH
+
+
+def get_auth_factory(connection_settings):
+    return {
+        BASIC_AUTH: BasicAuthFactory,
+        DIGEST_AUTH: DigestAuthFactory,
+        BEARER_AUTH: BearerAuthFactory
+    }[connection_settings.auth_type](connection_settings)
+
+
+class BearerProps(jsonobject.JsonObject):
+    token = jsonobject.StringProperty()
+    expiry = jsonobject.DateTimeProperty()
+    refresh_token = jsonobject.StringProperty()
+
+    @property
+    def is_valid(self):
+        return self.token and self.expiry and self.expiry > datetime.utcnow()
+
+
+class AuthFactory(AuthBase):
+    def __init__(self, connection_settings):
+        self.connection_settings = connection_settings
+        self._auth = None
+        self._prepare()
+
+    def _prepare(self):
+        pass
+
+    def __call__(self, r):
+        return self._auth(r)
+
+
+class BasicAuthFactory(AuthFactory):
+    def _prepare(self):
+        self._auth = HTTPBasicAuth(self.connection_settings.username, self.connection_settings.plaintext_password)
+
+
+class DigestAuthFactory(AuthFactory):
+    def _prepare(self):
+        self._auth = HTTPDigestAuth(self.connection_settings.username, self.connection_settings.plaintext_password)
+
+
+class BearerAuthFactory(AuthFactory):
```

Comment:
That all seems fine to me, I was really just wanting to illustrate storing the token for later use.

Reply:
:+1: Good call.

If we support OAuth1 -- and if I understand correctly -- I think we have to store the token, otherwise the user would have to re-authorize CommCare every time they want forms to be forwarded.

(I think the Repeater form allows users to select "OAuth1" but I haven't seen any code that implements that. When I have a chance, I'll look a bit harder, and if we don't actually support it, I'll drop the option.)